src/aliceVision/sfmData/colorize.cpp

```diff
@@ -31,97 +62,52 @@ void colorizeTracks(SfMData& sfmData)
   for(auto& landmarkPair : sfmData.getLandmarks())
     remainingLandmarksToColor.push_back(landmarkPair.second);
 
-  struct ViewInfo
-  {
-    ViewInfo(IndexT viewId, std::size_t cardinal)
-      : viewId(viewId)
-      , cardinal(cardinal)
-    {}
-
-    IndexT viewId;
-    std::size_t cardinal;
-    std::vector<std::reference_wrapper<Landmark>> landmarks;
-  };
-
-  std::vector<ViewInfo> sortedViewsCardinal;
-  sortedViewsCardinal.reserve(sfmData.getViews().size());
-  {
-    // create cardinal per viewId map
-    std::map<IndexT, std::size_t> viewsCardinalMap; // <ViewId, Cardinal>
-    for(const auto& landmarkPair : sfmData.getLandmarks())
-    {
-      const Observations& observations = landmarkPair.second.observations;
-      for(const auto& observationPair : observations)
-        ++viewsCardinalMap[observationPair.first]; // TODO: 0
-    }
-
-    // copy key-value pairs from the map to the vector
-    for(const auto& cardinalPair : viewsCardinalMap)
-      sortedViewsCardinal.push_back(ViewInfo(cardinalPair.first, cardinalPair.second));
-
-    // sort the vector, biggest cardinality first
-    std::sort(sortedViewsCardinal.begin(),
-              sortedViewsCardinal.end(),
-              [] (const ViewInfo& l, const ViewInfo& r) { return l.cardinal > r.cardinal; });
-  }
+  std::vector<LMColorAccumulator> landmarkInfo(sfmData.getLandmarks().size());
+  const Views& views = sfmData.getViews();
 
-  // assign each landmark to a view
-  for(ViewInfo& viewCardinal : sortedViewsCardinal)
+  #pragma omp parallel for
+  for( int viewIndex = 0; viewIndex < views.size(); ++viewIndex)
   {
-    std::vector<std::reference_wrapper<Landmark>> toKeep;
-    const IndexT viewId = viewCardinal.viewId;
+      image::Image<image::RGBColor> image;
```

Comment:
I cannot seem to pinpoint the reason, but when I load the image with RGBFColor, I'm able to access colors from the image as floating points, but the colors in landmarkInfo ends up as empty. Changing the colorspace has no effect on the result. It will take a couple of days to debug this one, as I suspect it has something to do with openmp. Any thoughts on the matter will be appreciated.

Reply:
Have you come to a conclusion on this?

Reply:
I have been trying for a week but cannot pinpoint the reason for this. Tried several changes with code and openmp functionality. I will give it a last try later today, and if it doesn't work, will stick with loading images as RGBColor

Reply:
@mugulmd Any thought on that?